packages/tools/fetch-tool/src/fluidFetchSnapshot.ts

```diff
@@ -248,7 +251,9 @@ async function saveSnapshot(name: string, fetchedData: IFetchedData[], saveDir:
         } else {
             // Write out same data for tree
             fs.writeFileSync(`${outDir}/${item.filename}.json`, data);
-            fs.writeFileSync(`${outDir}/decoded/${item.filename}.json`, JSON.stringify(JSON.parse(data), undefined, 2));
+            const decoded = bufferToString(buffer,"utf8");
```

Comment:
here

Reply:
@vladsud

Reply:
data was in base64, and outDir had files in base64, while files in decoded sub-folder were decoded and assumed to be utf-8 text.
That said here we are writing out a tree, and tree is not base64 encoded, it's just a tree (i.e. JSON in utf-8 format). We are not getting it via read/readBlob APIs, we are getting it via different API - getSnapshotTree, which returns a tree.

I have not traced code, but I believe isFetchedTree() check above differentiates blobs vs. tree. And the reason it's all in one map is because this tool puts all downloads (blobs and tree) into same map.

Please let me know if that was helpful, or not :)

Reply:
yes it's very helpful! i didn't know tree is in utf-8. thank you!